spec/src/main/asciidoc/configexamples.asciidoc

```diff
@@ -119,3 +119,66 @@ public class InjectedConfigUsageSample {
 }
 ----
 
+=== Aggregate related properties into a single POJO Example
+
+When injecting a number of related configuration properties, it can be tedious to repeat the statement of `ConfigProperty` in scatter places. 
+Since they are related, it makes more sense to aggregate them into a single property class. 
+
+MicroProfile Config provides a way to look up a number of configuration properties starting with the same prefix using the `@ConfigProperties` annotation.
+When annotating a class with `@ConfigProperties`, any of its fields, regardless of the visibility, maps to a configuration property via the following rules. 
+
+* Using `@ConfigProperty(name="<prop.name>")` to specify the configuration property name
+* Otherwise, 
+    ** If the `prefix` is present, the field `x` maps to the configuration property `<prefix>.x`. 
+    ** If the `prefix` is absent, the field `x` maps to the property name `x`. 
+
+Considering the following config source:
+
+[source, text]
+----
+config_ordinal = 120
+server.host = localhost
+server.port=9080
+server.endpoint=query
+oldServer.location=London
+----
+
+In order to retrieve the above properties in a single property class, you can use the `@ConfigProperties` annotation.
+
+[source, java]
+----
+
+@ConfigProperties
+public class ServerDetails {
+    public String host; 
+    public int port;   
+    private String endpoint; 
+    public @ConfigProperty(name="oldServer.location")
+    String location;  
+    public String getEndpoint() {
+        return endpoint;      
+    }     
+}
+----
+
+Then use the following to retrieve the properties with the specified prefix.
+
+* CDI Injection 
+
+[source, java]
+----
+@Inject @ConfigProperties("server") ServerDetails serverDetails;
+----
+
+The field `serverDetails.host` is assigned with the configuration value of `server.host`, `serverDetails.port` with the configuration value of `server.port`,
+`serverDetails.getEndpoint()` return the configuration value of `server.endpoint`, while the `serverDetails.location` populated with the configuration value of `oldServer.location`.
+
+
+* programmatically look up
+[source, java]
+----
+Config config = ConfigProvider.getConfig();
+ServerDetails serverDetails = config.getVale("server", ServerDetails.class);
```

Comment:
@radcortez see the above lookup for cdi and programmatic one. We don't need to add another method on `Config`.

Reply:
I get it, it still doesn't convince me :)

Because in `getValue(String, Class)`, the first argument is the full property name. Now, if you call it with a Config Properties class, is not the full property name anymore, but a prefix to other properties set in the class. Also, what if I don't have a prefix? Do I just call `getValue` with an empty String? 

I understand trying to save on the API and to reuse the method, but I think it is confusing. The argument might be same type, but the semantic meaning is different in each situation.

Reply:
@radcortez 

> Because in getValue(String, Class), the first argument is the full property name. Now, if you call it with a Config Properties class, is not the full property name anymore, but a prefix to other properties set in the class. 
> Also, what if I don't have a prefix? Do I just call getValue with an empty String?
If you don't have a prefix, you will supply an empty string. e.g
`config.getValue("", ServerDetails.class)`
There are cons/pros here. Let me list all of the options as a comment so that we can pick one.